iceberg/src/main/scala/org/apache/iceberg/spark/source/GpuIcebergPartitionReader.scala

```diff
@@ -0,0 +1,202 @@
+  }
+
+  private def newConf(): GpuIcebergParquetReaderConf = {
+    val nameMapping = Option(table.properties()
+      .get(TableProperties.DEFAULT_NAME_MAPPING))
+      .map(nm => NameMappingParser.fromJson(nm))
+
+    GpuIcebergParquetReaderConf(
+      task.cpuPartition.isCaseSensitive,
+      task.hadoopConf.value.value,
+      task.maxReadBatchSizeRows,
+      task.maxReadBatchSizeBytes,
+      task.gpuTargetBatchSizeBytes,
+      task.maxGpuColumnSizeBytes,
+      task.chunkedReaderEnabled,
+      task.maxChunkedReaderMemoryUsageSizeBytes,
+      task.parquetDebugDumpPrefix,
+      task.parquetDebugDumpAlways,
+      metrics,
+      threadConf,
+      task.expectedSchema,
+      nameMapping)
+  }
+
+  private def constantsMap(icebergFile: IcebergPartitionedFile): java.util.Map[Integer, _] = {
+    val task = tasks(icebergFile)
+    val requiredSchema = conf.expectedSchema
+    GpuIcebergPartitionReader.constantsMap(task, requiredSchema, table)
+  }
+}
+
+private object GpuIcebergPartitionReader {
+  def constantsMap(task: FileScanTask, readSchema: Schema, table: Table): JMap[Integer, _] = {
+    if (readSchema.findField(MetadataColumns.PARTITION_COLUMN_ID) != null) {
+      val partitionType: Types.StructType = Partitioning.partitionType(table)
+      PartitionUtil.constantsMap(task, partitionType, convertConstant)
+    }
+    else {
+      PartitionUtil.constantsMap(task, convertConstant)
+    }
+  }
+
+  private def convertConstant(`type`: Type, value: AnyRef): AnyRef = {
+    if (value == null) {
+      return null
+    }
+    `type`.typeId match {
+      case TypeID.DECIMAL =>
+        Decimal.apply(value.asInstanceOf[java.math.BigDecimal])
+      case TypeID.STRING =>
+        value match {
+          case utf8: Utf8 =>
+            UTF8String.fromBytes(utf8.getBytes, 0, utf8.getByteLength)
+          case _ =>
+            UTF8String.fromString(value.toString)
+        }
+      case TypeID.FIXED =>
+        value match {
+          case v: Array[Byte] => v
+          case fixed: GenericData.Fixed => fixed.bytes
+          case b => ByteBuffers.toByteArray(b.asInstanceOf[ByteBuffer])
+        }
+      case TypeID.BINARY =>
+        ByteBuffers.toByteArray(value.asInstanceOf[ByteBuffer])
+      case TypeID.STRUCT =>
+        val structType: Types.StructType = `type`.asInstanceOf[Types.StructType]
+        if (structType.fields.isEmpty) {
+          return new GenericInternalRow(Array.empty[Any])
+        }
+        val struct: StructLike = value.asInstanceOf[StructLike]
+
+        val values = structType.fields()
+          .asScala
+          .zipWithIndex
+          .map {
+            case (field, index) =>
+              val fieldType: Type = field.`type`
+              val value = struct.get(index, fieldType.typeId.javaClass).asInstanceOf[AnyRef]
+              convertConstant(fieldType, value).asInstanceOf[Any]
+          }
+          .toArray
+        new GenericInternalRow(values)
+      case _ => value
```

Comment:
From Spark org.apache.iceberg.data.IdentityPartitionConverters:
```scala
      case TIME:
        return DateTimeUtil.timeFromMicros((Long) value);
      case DATE:
        return DateTimeUtil.dateFromDays((Integer) value);
      case TIMESTAMP:
        if (((Types.TimestampType) type).shouldAdjustToUTC()) {
          return DateTimeUtil.timestamptzFromMicros((Long) value);
        } else {
          return DateTimeUtil.timestampFromMicros((Long) value);
        }
```
For date/time/timestamp, there is conversion, if we do not support, please throw exception.